sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -494,7 +494,8 @@ object QueryExecution {
   private[sql] def toInternalError(msg: String, e: Throwable): Throwable = e match {
     case e @ (_: java.lang.NullPointerException | _: java.lang.AssertionError) =>
       SparkException.internalError(
-        msg + " Please, fill a bug report in, and provide the full stack trace.",
+        msg + " You hit a bug in Spark or the Spark plugins you use. Please, report this bug " +
```

Comment:
Somewhat unrelated to the point of Spark vs. plugins, but at least for Spark, it seems we should provide guidance here on _how_ a user should report a bug.

Reply:
it's hard to give an answer here... stackoverflow, dev list, JIRA ticket, slack channels, etc.